web/packages/teleterm/electron-builder-config.js

```diff
@@ -142,6 +142,22 @@ module.exports = {
   },
   win: {
     target: ['nsis'],
+    sign: customSign => {
+      spawnSync(
+        'powershell',
+        [
+          '-noprofile',
+          '-executionpolicy',
+          'bypass',
+          '-c',
+          '$ProgressPreference = \'SilentlyContinue\'; ' +
+          '$ErrorActionPreference = \'Stop\'; ' +
+          '. ../../../build.assets/windows/build.ps1; ' +
+          `Invoke-SignBinary -UnsignedBinaryPath "${customSign.path}" -SignedBinaryPath "${customSign.resultOutputPath}"`,
```

Comment:
I think we have to do some changes:
1. Signing is invoked a few times, because electron-builder signs a few executables. We can add `"signingHashAlgorithms": ["sha256"]` [to avoid calling `sign` with `sha1` option](https://github.com/electron-userland/electron-builder/issues/3995#issuecomment-505725704) (which we ignore anyway). This will reduce the number of calls by half.
2. I think I was wrong when it comes to `resultOuputPath`, I see it is `undefined` (except of the last invocation https://github.com/gravitational/teleport.e/actions/runs/9211610782/job/25341406104). Idk what does this option mean, perhaps we should do `Invoke-SignBinary -UnsignedBinaryPath "${customSign.path}" -SignedBinaryPath "${customSign.path}"`, so the file is written in the same place. 
3. Signing should be done only on the CI, locally it throws `The term 'wsl-ubuntu-command' is not recognized as the name of a cmdlet`, should we add an env variable to control this? 🤔

Reply:
>Won't this potentially impact customers? If so, we might want to leave this as is.

It shouldn't, we don't support Windows 7 anyway.

>Do you know where I can find docs on the signing function/the object used for signing?
>If possible, I'd prefer to not output to the same file

There is no documentation for these options 🙈, that's all I could find: https://www.electron.build/configuration/win.html#how-do-delegate-code-signing
https://www.electron.build/tutorials/code-signing-windows-apps-on-unix.html#integrate-signing-with-electron-builder (take a look at the last example, they pass `path` to that JSign tool).

[I looked into the source code](https://github.com/electron-userland/electron-builder/blob/77f977435c99247d5db395895618b150f5006e8f/packages/app-builder-lib/src/codeSign/windowsCodeSign.ts#L69-L71), and it seems to me that you have to create the `resultOutputPath` manually. It is then used to rename the signed file again into the original file path. I believe it is for situations where the signing tool doesn't support signing in-place. I think it should be used like this:
```ts
    sign: customSign => {
      customSign.resultOutputPath = `signed-${customSign.path}`;
      ...
            `Invoke-SignBinary -UnsignedBinaryPath "${customSign.path}" -SignedBinaryPath "${customSign.resultOutputPath}"`,
        ],
        { stdio: 'inherit' }
      );
    }
```

BUT, if our tool is fine with having the same input and output path then I'd keep using `path`.

>Probably. Do I just need to do if (process.env["SOME VAR"]) { return; }?

Yeah, although it should rather be opt-in, not opt-out:
```ts
  if (!process.env.SOME_VAR) {
    console.warn(
      'missing $SOME_VAR: signing will be skipped'
    );
    return;
  }
```

Reply:
I think I've fixed the above. Please take a look. If the signing code is reasonable, then I'll run a test build prior to merge.

Reply:
I still think that we should set `"signingHashAlgorithms": ["sha256"]`. These signing algorithms are passed to the `sign` function, so the signing tool can be invoked with different algorithms (`customSign.hash`). We don't read that value at all, so we end up invoking the tool with exactly the same parameters twice (and I believe the second invocation will be signing the already signed binary).